include/bluetooth/scan.h

```diff
@@ -185,29 +185,97 @@ struct bt_scan_init_param {
 	const struct bt_le_conn_param *conn_param;
 };
 
+/**@brief Name filter status structure, used to inform application
+ *        which name filter is matched.
+ */
+struct bt_scan_name_filter_status {
+	/** Set to 1 if this type of filter is matched */
+	u8_t match : 1;
```

Comment:
I would simply use `bool` here. I don't think a bitfield makes sense since these structures are not packed and are also encapsulated in another non-packed structure.

Reply:
We used to have u8_t : 1 as type for bool in the SD because it is a binary, and the size of type bool is implementation defined.
So there is no reason here to continue to do so, and it might also generate more code to read and write it.

Reply:
Ok I changed this to bool type now.